docs/cheat-sheets/java-command-injection.mdx

```diff
@@ -0,0 +1,107 @@
+---
+slug: java-command-injection
+---
+
+import LinkToRegistryRule from "/src/components/LinkToRegistryRule"
+import CommandInjectionIntro from "/src/components/concept/_command-injection-intro.mdx"
+
+# Command injection prevention for Java
+
+<CommandInjectionIntro />
+
+#### Check your project for these conditions:
+
+```bash
+semgrep scan --config p/java-command-injection
+```
+
+---
+
+## 1. Running an OS command
+
+### 1.A. Running OS commands with Runtime.getRuntime().exec()
+
+The `exec` executes the specified string command in a separate process. This is dangerous if a command string is controlled by user input and could result in a command injection.
+
+Example:
+
+```java
+ // exec example
+Runtime.getRuntime().exec("ls -la")
+
+// Vunerable
+String input = "&& cat /etc/passwd"; // value supplied by user input
+Runtime r = Runtime.getRuntime();
+r.exec("some_tool -t param1 param2 " + input);
+
+// Vunerable
+String input = "cat /etc/passwd"; // value supplied by user input
+Runtime.getRuntime().exec("bash", "-c", input);
+```
+
+#### References
+
+- [`exec` documentation](https://docs.oracle.com/javase/7/docs/api/java/lang/Runtime.html#exec(java.lang.String))
+
+#### Mitigation
+
+- Always try to use internal Java API (if it exists) instead of running an OS command. In other words, use internal language features instead of invoking commands that can be exploited.
+- Do not include command arguments in a command string, use parameterization instead. For example:<br />
+    Use:
+    ```java
+    Runtime.getRuntime().exec("/path/to/myCommand", "myArg1", inputValue)
+    ```
+    Instead of:
+    ```java
+    Runtime.getRuntime().exec("bash", "-c", "myCommand myArg1 " + inputValue)
+    ```
+- If it is not possible, then strip the input of everything except alphanumeric characters provided for the command string and arguments.
+- Try to avoid non-literal values for the command string, such as `Runtime.getRuntime().exec("bash", "-c", input)` in the example above. Do not use `input` and similar functions.
+- If it is not possible, then do not let running arbitrary commands, use an allowlist for inputs.
+- Strip `!@#$;&*~"'{}][-+%^` characters from an input provided for command string. 
```

Comment:
`from user input that is incorporated in the command string which is later executed.`

Reply:
```suggestion
- Strip `!@#$;&*~"'{}][-+%^` characters from user input that is incorporated in the command string which is later executed.
```